### PR TITLE
Preston/serializers

### PR DESCRIPTION
**Pull request by pcoxcode**
Changed DEFUALT SETTINGS to Allow any so could post from anywhere, this will  need to be changed in the future.
Also I made the Answer Model have a question id placeholder as an ITNGER Field so we can check to see if the question Id matches the Question Asked

### CODE REVIEW

File: BQE_api/glossary/api/urls.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from django import urls
+from django.urls import path
+from django.urls.conf import include
+
+from rest_framework import routers, serializers, viewsets
+
+from .views import *
+
+app_name = 'glossary'
```
Comment (bransonsmith) [inline BQE_api/glossary/api/urls.py:9]: app_name not relevant here

File: BQE_api/glossary/api/urls.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+router = routers.DefaultRouter()
+router.register(r'word', WordViewSet)
+router.register(r'answer', AnswerViewSet)
+
+
+
+urlpatterns = [
+    path('', include(router.urls)),
+    
+
```
Comment (bransonsmith) [inline BQE_api/glossary/api/urls.py]: Lots of extra empty lines throughout this file can be cleaned up while you're in here too.

File: BQE_api/glossary/api/views.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    @action(detail=False, methods=['GET'], name='Random')
+    def random(self, request):
+        random_word_id = random.choice(list(self.queryset)).id
+        random_word = Word.objects.filter(id=random_word_id).first()
+        serializer = self.get_serializer(random_word)
+        return Response(serializer.data)
+        
+
+
+class AnswerViewSet(viewsets.ModelViewSet):
```
Comment (bransonsmith) [inline BQE_api/glossary/api/views.py:19]: Not sure what's going on here

File: BQE_api/glossary/models.py [hunk 4]
```diff
@@ -14,9 +14,9 @@ def __str__(self):
         return f'{self.word} {self.book} {self.chapter}: {self.verse}'
```
Comment (bransonsmith) [inline BQE_api/glossary/models.py]: We should get it working without Question/Question_Id first

File: BQE_api/glossary/urls.py [hunk 5]
```diff
@@ -2,17 +2,18 @@
 from django.urls import path
 from django.urls.conf import include
 from .views import *
-from rest_framework import routers, serializers, viewsets
+from glossary.api.urls import *
 
-router = routers.DefaultRouter()
-router.register(r'word', WordViewSet)
+app_name = 'glossary'
```
Comment (bransonsmith) [inline BQE_api/glossary/urls.py:7]: app_name not necessary here

Comment (bransonsmith) [reply BQE_api/glossary/urls.py:7]: We could have this variable, if we use it to generate routes below, but probably easier to just delete it for now

File: BQE_api/glossary/api/serializers.py [hunk 6]
```diff
@@ -11,3 +12,8 @@ class Meta:
         model = Entry
         fields = ['id', 'word', 'book', 'chapter', 'verse']
 
+class AnswerSerializer(serializers.ModelSerializer):
```
Comment (bransonsmith) [inline BQE_api/glossary/api/serializers.py:14]: We need to get it working without question id first

File: BQE_api/glossary/api/urls.py (hunk 1 above)
Comment (pcoxcode) [reply BQE_api/glossary/api/urls.py:9]: it is for routers

**Comment by bransonsmith** [comment]
With what we know now, this task is too big for 1 PR.
We just have a lot of different pieces that each need independent research.

I am closing this PR. Leave this branch around as a reference to grab code snippets from until we get all the new PRs into master.
I will write up new, smaller tasks in Trello.

**Pull request closed without merging**